### ISSUE TITLE
Some troubles found in version 3.0.0-beta-2

### ISSUE THREAD

**Issue by BnitoBzh** [sir-dunxalot/ember-tooltips#253]
In this file : https://github.com/sir-dunxalot/ember-tooltips/blob/3.0.0-beta.2/addon/components/ember-tooltip-base.js

- `capitalize` method fails with placement variants (-start, -end) because getOppositeSide method does not handle these cases.
I think getOppositeSide method must be like this : 
```
function getOppositeSide(side) {
  side = side.split('-');

  switch (side) {
    case 'top':
      side[0] = 'bottom';
      break;
    case 'right':
      side[0] = 'left';
      break;
    case 'bottom':
      side[0] = 'top';
      break;
    case 'left':
      side[0] = 'right';
      break;
  }

  return side.join('-');
}
```

- At line 361, i think there is double call from the tooltip show method. `tooltip.show();` is not necessary because `this.show();` do the job.

- During my tests on my application, the willDestroy throw an error because `this.sendAction('onDestroy', this);` can not work when this is null.

- And i found that the `onUpdate` method (popperOptions) is always called after tooltip destroy, i don't know why (for test it, juste add `console.log('test')` in the `setSpacing` method. It is not normal because the browser continue to calculate the spacing of all tooltips after their deletion ....

**Comment by mattmarcum** [comment]
👍 The `this.sendAction('onDestroy', this);` is a deal breaker, I can't upgrade if I can't get tests to pass.

**Comment by maxfierke** [comment]
The bottom three have been addressed as of 3.0.0-beta.3.

**Comment by sir-dunxalot** [comment]
I have created #281 to track the first bullet (placement variants) since that may not be fixed until `3.0.0` is released.

**Comment by sir-dunxalot** [comment]
Fixed by #272.
